src/lib/hooks/useDomain.ts

```diff
@@ -0,0 +1,56 @@
+//- React Imports
+import { useQuery } from 'react-query';
+
+//- Hooks Imports
+import { useZnsSdk } from './useZnsSdk';
+
+//- Library Imports
+import { Domain } from '@zero-tech/zns-sdk/lib/types';
+
+export interface UseDomainReturn {
+	domain: Domain | undefined;
+	error: any;
+	isError: boolean;
+	isLoading: boolean;
+	isSuccess: boolean;
+}
+
+export const useDomain = (domainId: string): UseDomainReturn => {
+	// SDK
+	const sdk = useZnsSdk();
+
+	// Query
+	const {
+		error,
+		isError,
+		isLoading,
+		isSuccess,
+		data: { domain } = {},
+	} = useQuery(
+		`domain-${domainId}`,
+		async () => {
+			try {
+				const domain = await sdk.getDomainById(domainId);
+
+				return {
+					domain,
+				};
+			} catch (error: any) {
+				throw new Error(error);
+			}
+		},
+		{
+			retry: false,
+			refetchOnMount: false,
+			refetchOnWindowFocus: false,
```

Comment:
Thinking about  `refetchOnMount` and `refetchOnWindowFocus` true value will be helpful to sync domain data for this hook. Not sure...